apps/prairielearn/src/tests/fileEditor.test.ts

```diff
@@ -320,57 +327,36 @@ describe('test file editor', function () {
     after('shut down testing server', helperServer.after);
 
     describe('disallow edits inside exampleCourse', function () {
-      badGet(badExampleCoursePathUrl);
+      badGet(badExampleCoursePathUrl, 403, true);
     });
   });
 });
 
-function badGet(url) {
+function badGet(url, expected_status, should_parse) {
   describe(`GET to edit url with bad path`, function () {
-    it('should not load successfully', function (callback) {
+    it(`should load with status ${expected_status}`, function (callback) {
       locals.preStartTime = Date.now();
-      request(url, function (error, response) {
+      request(url, function (error, response, body) {
         if (error) {
           return callback(error);
         }
         locals.postStartTime = Date.now();
-        if (response.statusCode !== 400) {
+        if (response.statusCode !== expected_status) {
           return callback(new Error('bad status: ' + response.statusCode));
         }
+        page = body;
         callback(null);
       });
     });
-  });
-}
-
-function badPost(action, fileEditContents, url) {
-  describe(`POST to edit url with action ${action} and with bad path`, function () {
-    it('should not load successfully', function (callback) {
-      const form = {
-        __action: action,
-        __csrf_token: locals.__csrf_token,
-        file_edit_contents: b64Util.b64EncodeUnicode(fileEditContents),
-        file_edit_user_id: locals.file_edit_user_id,
-        file_edit_course_id: locals.file_edit_course_id,
-        file_edit_dir_name: '../PrairieLearn/',
-        file_edit_file_name: 'config.json',
-        file_edit_orig_hash: locals.file_edit_orig_hash,
-      };
```

Comment:
This test was checking the case where `file_edit_dir_name` and `file_edit_file_name` were manipulated independently of the URL. Now that we always determine which file we're editing from the URL, this test doesn't make sense.

I thought about writing a new test to cover a similar case (e.g., constructing a url that tries to edit `../../../PrairieLearn/config.json`, but ran into two problems with that:

- Every client I tried, including `fetch()`, will resolve `..` before sending the request
- Even if we could work around that (of course _some_ client will let us avoid path normalization), we'd never be able to get a CSRF token for a subsequent POST because the GET handler will refuse to serve the edit form for a bad path in the first place! We could always generate our own CSRF since we know the secret... but it feels like we'd be testing a scenario that could never actually happen in prod.

At any rate, seemed like more trouble than its worth, though I'll figure out a wayto write this test if someone feels strongly about it.

Reply:
I'm fine with just not testing this explicitly.